tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/routes/DownloadAllRoutes.scala

```diff
@@ -0,0 +1,190 @@
+                                  val sessionTranslator: SessionTranslator,
+                                  val messageIdManager: MessageIdManager,
+                                  val messageIdFactory: MessageId.Factory,
+                                  val metricFactory: MetricFactory) extends JMAPRoutes {
+
+  private val accountIdParam: String = "accountId"
+  private val emailIdParam: String = "emailId"
+  private val nameParam: String = "name"
+  private val downloadUri = s"/downloadAll/{$accountIdParam}/{$emailIdParam}"
+
+  override def routes(): stream.Stream[JMAPRoute] = Stream.of(
+    JMAPRoute.builder
+      .endpoint(new Endpoint(HttpMethod.GET, downloadUri))
+      .action(this.get)
+      .corsHeaders,
+    JMAPRoute.builder
+      .endpoint(new Endpoint(HttpMethod.OPTIONS, downloadUri))
+      .action(JMAPRoutes.CORS_CONTROL)
+      .noCorsHeaders)
+
+  private def get(request: HttpServerRequest, response: HttpServerResponse): Mono[Void] =
+    SMono(authenticator.authenticate(request))
+      .flatMap(mailboxSession => getIfOwner(request, response, mailboxSession))
+      .onErrorResume {
+        case _: ForbiddenException | _: AccountNotFoundException =>
+          respondDetails(response, ProblemDetails(status = FORBIDDEN, detail = "You cannot download in others accounts"))
+        case e: UnauthorizedException =>
+          LOGGER.warn("Unauthorized", e)
+          respondDetails(e.addHeaders(response), ProblemDetails(status = UNAUTHORIZED, detail = e.getMessage))
+        case _: BlobNotFoundException =>
+          respondDetails(response, ProblemDetails(status = NOT_FOUND, detail = "The resource could not be found"))
+        case e =>
+          LOGGER.error("Unexpected error upon download {}", request.uri(), e)
+          respondDetails(response, ProblemDetails(status = INTERNAL_SERVER_ERROR, detail = e.getMessage))
+      }
+      .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)
+      .asJava()
+      .`then`
+
+  private def get(request: HttpServerRequest, response: HttpServerResponse, mailboxSession: MailboxSession): SMono[Unit] =
+    SMono.just(messageIdFactory.fromString(request.param(emailIdParam)))
+      .flatMap(messageId => SFlux(messageIdManager.getMessagesReactive(ImmutableList.of(messageId), FetchGroup.HEADERS_WITH_ATTACHMENTS_METADATA, mailboxSession)).singleOrEmpty())
+      .flatMapMany(messageResult => SFlux.fromIterable(messageResult.getLoadedAttachments.asScala))
+      .flatMap(attachment => BlobId.of(attachment.getAttachmentId.getId)
+        .fold(e => SMono.error(e),
+          blobResolvers.resolve(_, mailboxSession)))
+      .collectSeq()
+      .flatMap(blobs => downloadBlob(
+        optionalName = queryParam(request, nameParam),
+        response = response,
+        blobContentType = ContentType.of("zip"),
+        blobs)
+        .`then`())
+
+  private def getIfOwner(request: HttpServerRequest, response: HttpServerResponse, mailboxSession: MailboxSession): SMono[Unit] =
```

Comment:
`getIfOwner` as a method name but it calls `delegateIfNeeded` under the hood.

The two kind of contradicts.

Maybe we can rename this method to getWithDeletation or something like that?